### PR TITLE
CI: add branch pattern 'pr/**' to PR test workflow

### PR DESCRIPTION
**Pull request by ansiwen**
In order to run CI jobs also on PRs that are based on other PRs, this
change adds the pattern 'pr/**' to the tested base branches.  That
means, if PRs are pushed that are based on branches that start with
pr/... these PRs are also get tested. So, if we - by convention - push
PRs to branches like for example pr/ansiwen/myfix42, then all PRs
that are not based on master but another PR get tested as well.

For reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

Signed-off-by: Sven Anderson <sven@redhat.com>

### CODE REVIEW

**Comment by nixpanic** [comment]
Is that `pr/` prefix required? I can not find it in the document you linked... If it is a convention you are recommending for this repository, it should probably get documented in some developer guide?

**Comment by ansiwen** [comment]
> Is that `pr/` prefix required? I can not find it in the document you linked... If it is a convention you are recommending for this repository, it should probably get documented in some developer guide?

@nixpanic It's not required, I am proposing that. We can document it somewhere, do you have a suggestion? But I guess we will not allow public push access to the go-ceph repo, so that is only important for official developers of go-ceph anyway.

**Comment by phlogistonjohn** [comment]
Just to make sure I understand this:
IF a branch is based on another branch with the naming convention where the base branch starts with "pr/" it will test the current pr rather than ignoring it. If that's true, I'm OK with this. Maybe a comment with a terse description of this would be good.

**Comment by ansiwen** [comment]
> Just to make sure I understand this:
> IF a branch is based on another branch with the naming convention where the base branch starts with "pr/" it will test the current pr rather than ignoring it. If that's true, I'm OK with this. Maybe a comment with a terse description of this would be good.

Yes, you understood right. I added a comment and rephrased the commit message. Hope it's clearer now.

**Review by phlogistonjohn — DISMISSED** [review]

**Review by nixpanic — COMMENTED** [review]
I still don't really get how this works... How do branches need to be structured?

If I push my branch that is based on an other one, would I not need to push the other branch as well? Or is there some magic that prevents the other branche's PR to not go out of sync?

Also, typo in the commit message `by convetion`.

**Comment by ansiwen** [comment]
@Mergifyio rebase

**Review by phlogistonjohn — APPROVED** [review]

**Comment by ansiwen** [comment]
@Mergifyio rebase

**Comment by ansiwen** [comment]
> I still don't really get how this works... How do branches need to be structured?
> 
> If I push my branch that is based on an other one, would I not need to push the other branch as well? Or is there some magic that prevents the other branche's PR to not go out of sync?

(Amost) no magic involved: let's say you have two branches, A and B. B depends on A, and therefor is based on it. You push both independently to the go-ceph repo as `pr/A` and `pr/B`. You create the PR for A as usual. When you create the PR for B you don't select `master` but `pr/A` as the base branch. Normally the CI wouldn't get triggered for this PR, because it's not based on master. With this PR it would also get tested, because the base is `pr/A` and matches the `pr/**` pattern. Then you proceed with reviewing both PRs, and because PR B is based on A, you only see the commits that actually belong to B. Whenever the branch `pr/A` changes, you have to rebase `pr/B`, exactly like you also have to rebase `pr/A` whenever `master` changes. So no additional work, but also not less. The only magic involved is: when any branch gets merged that is the base for other PRs and then gets deleted, then all these PRs are rebased to the branch the old base has been merged into. So when `pr/A` gets merged into `master` and then deleted, then the PR B will automatically be rebased to `master`. In order to avoid, that `pr/B` gets accidentally merged into `pr/A` earlier, you can write `Depends on #number_of_pr_A` and the dpulls service will block merging, until PR A got merged. 

> Also, typo in the commit message `by convetion`.

Thanks, will fix.

**Comment by phlogistonjohn** [comment]
@ansiwen @nixpanic aside from the spelling fix for the commit message, where do we stand on this?

**Review by phlogistonjohn — DISMISSED** [review]

**Comment by ansiwen** [comment]
I suggest we could have a bit more agility with PRs like this, that are easily revertible and help with productivity, and require only one review?

**Review by phlogistonjohn — APPROVED** [review]
I'm fairly sure I approved this before, but it must have gotten dismissed by mergify or something.
While I don't plan on using it myself right away, it seems pretty clear that it won't disrupt my existing workflows. Thus, I agree that we shouldn't be too picky about this.

**Comment by phlogistonjohn** [comment]
Given the above let's give it until Monday (March 8) and if there still are no 2nd approvals I'll merge anyway.

**Review by agarwal-mudit — APPROVED** [review]

**Comment by agarwal-mudit** [comment]
I guess this anyway needs manual merge because it is a change in workflow.

**Pull request merged**
